### PR TITLE
Add support for selectors dict concatenation

### PR DESCRIPTION
**Pull request by AlessandroPatti**
Currently it is possible to concatenate multiple select like
`[...] + select({"...": [...]}) + select({"...": [...]})`
However it is not possible to perform a concatenation when the attribute type is a dictionary.
This add support for the syntax:
`{...} + select({"...": {"...": "..."}}) + select({"...": {"...": "..."}})`
All the dictionaries are merged in order, so that each dictionary overrides the keys of the predecessors if the key is already present.

### CODE REVIEW

**Comment by laurentlb** [comment]
Since we don't allow dict concatenation, I don't think we want to add this construct.

**Comment by gregestren** [comment]
@laurentlb - is there a link to that policy for clarification?

**Comment by aiuto** [comment]
I am asserting that this is a good enough link: https://docs.bazel.build/versions/master/skylark/lib/dict.html

TL;DR; Starlark follows python for basic types and python does not allow dict + dict.
Now, we should have a Starlark language definition that is more precise, but that is a different issue.
